ui/component/viewers/videoViewer/internal/videojs.jsx

```diff
@@ -538,6 +541,12 @@ export default React.memo<Props>(function VideoJs(props: Props) {
         document.querySelector('.video-js-parent')?.append(window.oldSavedDiv);
       }
 
+      // need this for canvas to work
+      const player = document.querySelector('video.vjs-tech');
+      if (player) player.setAttribute('crossorigin', 'anonymous');
+
+      vjsPlayer.snapshotButton({ fileTitle: title });
```

Comment:
Only thing I found to differentiate an audio file that runs "as a video" with a poster is to check `player.tech_.sourceHandler_` (if it's audio only it's undefined) but I don't like it much versus just doing an if-else block here

Reply:
Should follow the framework method and not the convenient path, though.

Would passing our claim-based `isAudio` as the plugin options and handle things there work?  The "skip for mobile" handling is already done there, so why split the logic to several places.

Reply:
I think it would be cool to open-source this as a standalone plugin as the other options are abandoned and I wasn't able to get them working, so from that perspective I don't want to pass a second argument that would only really apply in our case, and using the `player.tech_.sourceHandler_` API seems hacky and I don't have much faith in it, plus it's unclear for other devs versus the `isAudio` boolean value. We already do a check for `IS_MOBILE` to conditionally run the `vttThumbnails` plugin so that's why I don't feel uncomfortable using this design. (`ui/component/viewers/videoViewer/internal/videojs.jsx:515` on `master`)

![Screen Shot 2022-10-21 at 3 00 51 PM](https://user-images.githubusercontent.com/7200471/197201973-f06e8b64-a2cf-4b5a-9358-91211f2d542e.JPG)